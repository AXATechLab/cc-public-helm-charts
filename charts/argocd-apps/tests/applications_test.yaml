suite: test argocd-apps applications
templates:
  - templates/applications.yaml
chart:
  version: 1.0.0
release:
  name: my-release
tests:
  - it: should have correct applications
    set:
      environmentName: test
      applications:
        demo-fastify:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
          parameters:
            app.image.tag: '$ARGOCD_APP_REVISION'
            test.enabled: true
        authoringtool-frontend:
          repoURL: 'https://github.com/AXATechLab/cc-authoring-client'
          branch: "feature/new-pipeline"
          environmentManagerLogo: 'authoringtool-frontend.png'
          autoSync: true
          enableEnvSuffix: false
          parameters:
            app.image.tag: '$ARGOCD_APP_REVISION'
        credentials:
          repoURL: 'https://github.com/AXATechLab/cc-applications'
          path: preprod/credentials
          extraValuesFiles:
            - values-prod.yaml
    asserts:
      - matchSnapshot: {}
  - it: should have correct sync policy with default true
    set:
      default:
        autoSync: true
      applications:
        demo-fastify:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
          autoSync: false
        demo-fastify-2:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
    asserts:
      - matchSnapshot: {}
  - it: should have correct sync policy with default false
    set:
      default:
        autoSync: false
      applications:
        demo-fastify:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
          autoSync: true
        demo-fastify-2:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
    asserts:
      - matchSnapshot: {}
  - it: should have correct label based on application alias
    set:
      environmentName: "applications"
      applications:
        llm-claims-api:
          alias: monorepo
          namespace: tool-prod-axa-rev
          project: tool-prod-axa-rev
          branch: main
          repoURL: 'https://github.com/AXATechLab/cc-applications'
          path: "namespaces/dev-test/dependencies"
          valuesFiles:
            - values.yaml
            - values-branches.yaml
    asserts:
      - equal:
          path: metadata.labels["app.kubernetes.io/application-name"]
          value: "monorepo"
  - it: should have correct branch name label
    set:
      applications:
        authoringtool-frontend:
          branch: "feat/ab-14371-link-to-rule-editor-always-goes-to-first-rule-no-focusnode"
    asserts:
      - equal:
          path: metadata.labels["app.kubernetes.io/branch"]
          value: "feat-ab-14371-link-to-rule-editor-always-goes-to-first-rule-no"
  - it: should render valuesObject correctly
    set:
      applications:
        demo-fastify:
          repoURL: 'https://github.com/AXATechLab/demo-fastify'
          valuesObject:
            common:
              env:
                DATABASE_DB: qa
            authentication:
              env:
                some: value
    asserts:
      - equal:
          path: spec.source.helm.valuesObject
          value: 
            authentication:
              env:
                some: value
            common:
              env:
                DATABASE_DB: qa
