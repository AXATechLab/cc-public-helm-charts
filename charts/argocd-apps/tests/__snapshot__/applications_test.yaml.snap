should have correct applications:
  1: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: authoringtool-frontend
        app.kubernetes.io/branch: feature-new-pipeline
        app.kubernetes.io/environment-name: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        environment-manager/logo: authoringtool-frontend.png
        helm.sh/chart: argocd-apps-1.0.0
      name: authoringtool-frontend-test
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: test
            - name: global.branch
              value: feature-new-pipeline
            - name: global.applicationName
              value: authoringtool-frontend
            - name: app.image.tag
              value: $ARGOCD_APP_REVISION
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/cc-authoring-client
        targetRevision: feature/new-pipeline
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
  2: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: credentials
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: credentials-test
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: test
            - name: global.branch
              value: main
            - name: global.applicationName
              value: credentials
          valueFiles:
            - values.yaml
            - values-prod.yaml
        path: preprod/credentials
        repoURL: https://github.com/AXATechLab/cc-applications
        targetRevision: main
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
  3: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: demo-fastify
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: demo-fastify-test
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: test
            - name: global.branch
              value: main
            - name: global.applicationName
              value: demo-fastify
            - name: app.image.tag
              value: $ARGOCD_APP_REVISION
            - name: test.enabled
              value: "true"
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/demo-fastify
        targetRevision: main
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
should have correct sync policy with default false:
  1: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: demo-fastify
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: demo-fastify-my-release
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: my-release
            - name: global.branch
              value: main
            - name: global.applicationName
              value: demo-fastify
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/demo-fastify
        targetRevision: main
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
  2: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: demo-fastify-2
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: demo-fastify-2-my-release
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: my-release
            - name: global.branch
              value: main
            - name: global.applicationName
              value: demo-fastify-2
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/demo-fastify
        targetRevision: main
should have correct sync policy with default true:
  1: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: demo-fastify
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: demo-fastify-my-release
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: my-release
            - name: global.branch
              value: main
            - name: global.applicationName
              value: demo-fastify
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/demo-fastify
        targetRevision: main
  2: |
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      labels:
        app.kubernetes.io/application-name: demo-fastify-2
        app.kubernetes.io/branch: main
        app.kubernetes.io/environment-name: my-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: argocd-apps-1.0.0
      name: demo-fastify-2-my-release
    spec:
      destination:
        namespace: tool-prod-axa-rev
        server: https://kubernetes.default.svc
      project: tool-prod-axa-rev
      source:
        helm:
          parameters:
            - name: global.environmentName
              value: my-release
            - name: global.branch
              value: main
            - name: global.applicationName
              value: demo-fastify-2
          valueFiles:
            - values.yaml
        path: helm
        repoURL: https://github.com/AXATechLab/demo-fastify
        targetRevision: main
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
